catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/data/impl/MetacardImpl.java

```diff
@@ -157,6 +157,9 @@ public MetacardImpl(Metacard metacard, MetacardType type) {
             throw new IllegalArgumentException(
                     MetacardType.class.getName() + " instance should not be null.");
         }
+        if (metacard.getSourceId() != null) {
```

Comment:
The `readObject` and `writeObject` methods should be analyzed to ensure they are not broken. Also, by serializing the cloned metacard you should be able to verify your change.